lib/config/hse_gparams.c

```diff
@@ -103,15 +104,21 @@ logging_destination_jsonify(const struct param_spec *const ps, const void *const
     }
 }
 
-void
+static void
 socket_path_default(const struct param_spec *ps, void *value)
 {
-    assert(ps);
-    assert(value);
-
+    const char *dir;
     HSE_MAYBE_UNUSED int n;
 
-    n = snprintf(value, sizeof(hse_gparams.gp_socket.path), "/tmp/hse-%d.sock", getpid());
+    INVARIANT(ps);
+    INVARIANT(value);
+
+    dir = getenv("XDG_RUNTIME_DIR");
+    if (!dir || strlen(dir) == 0)
```

Comment:
*What do you guys thing of the strlen() check here? Unnecessary?

Reply:
From https://specifications.freedesktop.org/basedir-spec/basedir-spec-latest.html

> All paths set in these environment variables must be absolute. If an implementation encounters a relative path in any of these variables it should consider the path invalid and ignore it.

You  could use:   `if (!dir || *dir != '/')`

Reply:
Great catch.